test/cmdlineTests/standard_irOptimized_requested/output.json

```diff
@@ -33,6 +33,7 @@ object \"C_6\" {
                 }
                 default { }
             }
+            if iszero(calldatasize()) { }
```

Comment:
This is optimized IR output - why isn't this removed?

Reply:
This does not look like optimized code in general. Maybe something is wrong with the optimizer there? The function `fun_f_5` in the constructor, for example, should be removed.

Reply:
Ah, ok, the test is a bit confusing - it requests "irOptimized" via standard json, but does not actually enable yul optimization, so the code is in fact unoptimized... Should we run the yul optimizer by default, if "irOptimized" is requested? It's an issue separate from this PR in any case.